src/string.cr

```diff
@@ -3685,32 +3725,20 @@ class String
   # "Dizzy Miss Lizzy".byte_index('z', -4)  # => 13
   # "Dizzy Miss Lizzy".byte_index('z', -17) # => nil
   # ```
-  def byte_index(char : Char, offset = 0) : Int32?
-    return byte_index(char.ord, offset) if char.ascii?
+  def byte_index(search : Char, offset = 0) : Int32?
```

Comment:
Renaming the external parameter name would be a breaking change. (`string.byte_index(char: c)` would not compile anymore.
I suppose it might make sense to unify the argument names in the API, but that should be a separate change and not in this PR.
For now I'd suggest to keep at least the external name as is.
```suggestion
  def byte_index(char search : Char, offset = 0) : Int32?
```